### PR TITLE
CMR-6798 - Added the Granule Search, based on the Collection Search

### PR DESCRIPTION
**Pull request by jceaser**
## Description
Adds a Granule Search API based on the Collection API

## Testing
```
import cmr.search.granule as gran
params = {'provider': 'ORNL_DAAC'}
results = gran.search(params)
print("Found {} records.".format(len(results)))
for i in results:
    print (i)
```

### CODE REVIEW

**Review by mitchstartzel — APPROVED** [review]

File: CMR/python/cmr/search/granule.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import cmr.search.common as scom
+
+# ******************************************************************************
+# filter function lambdas
+
+def all_fields(item):
+    """ Pass through, do no filtering - is this needed outside of testing?"""
+    return scom.all_fields(item)
+def meta_fields(item):
+    """return only the the meta objects"""
```
Comment (jaybarra) [inline CMR/python/cmr/search/granule.py:46]: Same as the other PR. Let's be consistent with capitalization of first word of docstrings.

File: CMR/python/cmr/search/granule.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html
+"""
+
+import cmr.search.common as scom
+
+# ******************************************************************************
+# filter function lambdas
+
+def all_fields(item):
+    """ Pass through, do no filtering - is this needed outside of testing?"""
```
Comment (jaybarra) [inline CMR/python/cmr/search/granule.py:43]: Turn the question into an imperative instead.

**Review by jaybarra — APPROVED** [review]
Looks ok except for capitalizations on docstrings

File: CMR/python/cmr/search/granule.py (hunk 1 above)
Comment (jceaser) [reply CMR/python/cmr/search/granule.py:46]: will do

Comment (jceaser) [reply CMR/python/cmr/search/granule.py:46]: search: """\s?([a-z])
replace: """ \u\1

Comment (jceaser) [reply CMR/python/cmr/search/granule.py]: had to add some files to the request, but did a wider search, marking as resolved

File: CMR/python/cmr/search/granule.py (hunk 2 above)
Comment (jceaser) [reply CMR/python/cmr/search/granule.py]: resulted in adding some files with similar comments, marking an resolved.

**Pull request merged**